src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -581,32 +512,32 @@ private function registerDataCollectorConfiguration(ContainerBuilder $container,
         }
     }
 
-    private function registerMercureConfiguration(ContainerBuilder $container, array $config, XmlFileLoader $loader, bool $useDoctrine)
+    private function registerMercureConfiguration(ContainerBuilder $container, array $config, XmlFileLoader $loader): void
     {
-        if (!$config['mercure']['enabled']) {
+        if (!$this->isConfigEnabled($container, $config['mercure'])) {
             return;
         }
 
         $loader->load('mercure.xml');
         $container->getDefinition('api_platform.mercure.listener.response.add_link_header')->addArgument($config['mercure']['hub_url'] ?? '%mercure.default_hub%');
 
-        if ($useDoctrine) {
+        if ($this->isConfigEnabled($container, $config['doctrine'])) {
             $loader->load('doctrine_orm_mercure_publisher.xml');
         }
     }
 
-    private function registerMessengerConfiguration(array $config, XmlFileLoader $loader)
+    private function registerMessengerConfiguration(ContainerBuilder $container, array $config, XmlFileLoader $loader): void
     {
-        if (!$config['messenger']['enabled']) {
+        if (!$this->isConfigEnabled($container, $config['messenger'])) {
             return;
         }
 
         $loader->load('messenger.xml');
     }
 
-    private function registerElasticsearchConfiguration(ContainerBuilder $container, array $config, XmlFileLoader $loader)
+    private function registerElasticsearchConfiguration(ContainerBuilder $container, array $config, XmlFileLoader $loader): void
     {
-        $enabled = $config['elasticsearch']['enabled'] && class_exists(Client::class);
```

Comment:
I removed this check because `elasticsearch` support needs to be explicitly enabled, so it's more useful for the user to get an exception in this case, instead of silently disabling support.

Reply:
Maybe better if we just throw an exception here directly. WDYT @meyerbaptiste?

Reply:
Changed to throw exception in `Configuration` instead.